src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +387,160 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
+                        _closeDialog(true);
+                    });
+
+                    InsertIntoTextBuffer(document, libraryInstallationState);
+
+                    // Save manifest file so we can restore library files.
+                    rdt.SaveFileIfDirty(configFilePath);
+                }
             }
             catch { }
         }
 
+        private void InsertIntoTextBuffer(IVsTextBuffer document, LibraryInstallationState libraryInstallationState)
+        {
+            ITextBuffer textBuffer = GetTextBuffer(document);
+
+            if (textBuffer != null)
+            {
+                string insertionText;
+
+                JSONMember libraries = GetLibraries(textBuffer);
+
+                if (libraries != null && libraries.Children != null)
+                {
+                    JSONArray jsonArray = libraries.Children.OfType<JSONArray>().First();
+
+                    string newLibrary = GetLibraryToBeInserted(libraryInstallationState);
+                    bool containsLibrary = jsonArray.BlockItemChildren.Any();
+
+                    if (containsLibrary)
+                    {
+                        insertionText = _commaDelimiter + Environment.NewLine + newLibrary + Environment.NewLine;
+                    }
+                    else
+                    {
+                        insertionText = newLibrary + Environment.NewLine;
+                    }
+
+                    int insertionIndex = libraries.AfterEnd - 1;
+
+                    if (insertionIndex > 0)
+                    {
+                        FormatSelection(textBuffer, insertionIndex, insertionText);
+                    }
+                }
+            }
+        }
+
+        private void FormatSelection(ITextBuffer textBuffer, int insertionIndex, string insertionText)
+        {
+            Controller.TextViewData viewData = Controller.TextViewConnectionListener.GetTextViewDataForBuffer(textBuffer);
+            ITextView textView = viewData.LastActiveView;
+
+            using (ITextEdit textEdit = textBuffer.CreateEdit())
+            {
+                textEdit.Insert(insertionIndex, insertionText);
+                textEdit.Apply();
+
+                IOleCommandTarget commandTarget = Shell.Package.GetGlobalService(typeof(Shell.Interop.SUIHostCommandDispatcher)) as IOleCommandTarget;
+
+                SendFocusToEditor(textView);
+
+                SnapshotSpan snapshotSpan = new SnapshotSpan(textView.TextSnapshot, insertionIndex, insertionText.Length + 1);
+                textView.Selection.Select(snapshotSpan, false);
+
+                Guid guidVSStd2K = VSConstants.VSStd2K;
+
+                commandTarget.Exec(
+                    ref guidVSStd2K,
+                    (uint)VSConstants.VSStd2KCmdID.FORMATSELECTION,
+                    (uint)OLECMDEXECOPT.OLECMDEXECOPT_DODEFAULT,
+                    IntPtr.Zero,
+                    IntPtr.Zero);
+
+                textView.Selection.Clear();
+            }
```

Comment:
>} [](start = 12, length = 1)

Does disposing after all the format selection make this a single edit? (IE, I'm trying to understand why the using block isn't closed earlier)

Reply:
The using can be closed earlier. I see two entries in undo stack:

Format 
TextBufferChange

Updated for version 10

---
In reply to: [197933019](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197933019) [](ancestors = 197933019)